server/src/main/java/org/apache/iotdb/db/service/RegisterManager.java

```diff
@@ -64,4 +65,21 @@ public void deregisterAll() {
     iServices.clear();
     logger.info("deregister all service.");
   }
+  
+  /**
+   * stop all service and clear iService list.
+   */
+  public void shutdownAll() {
+    //we stop JMXServer at last
+    Collections.reverse(iServices);
+    for (IService service : iServices) {
+      try {
+        service.shutdown(10000);
+      } catch (Exception e) {
+        logger.error("Failed to stop {} because:", service.getID().getName(), e);
```

Comment:
I think you should rethrow the exception, and in the tests, you should catch it, and add failed() clause.

Reply:
Very good point! Thanks.